### DISCUSSION TITLE
Setting two 1-1 relations from separate tables

### DISCUSSION THREAD

**Discussion by spidertwin2** [prisma/prisma#7517] [Q&A] [answered]
Hello,

I want to store `authorName` in `Post`.  But that value is stored in `Profile` model's `name` field.

How should I map this relation in the schema such that the current user's `name` is automatically retrieved from `Profile` and stored in `authorName`?   Just as `authorId` is automatically assigned.

```prisma
model User {
  id             Int            @id @default(autoincrement())
  createdAt      DateTime       @default(now())
  updatedAt      DateTime       @updatedAt
  email          String         @unique
  hashedPassword String?
  profile        Profile?
  posts          Post[]
}

model Profile {
  id         Int       @id @default(autoincrement())
  name       String?
  user       User      @relation(fields: [userId], references: [id])
  userId     Int       @unique
}

model Post {
  id         Int     @id @default(autoincrement())
  title      String?
  body       String?
  published  Boolean @default(false)
  author     User?   @relation(fields: [authorId], references: [id])
  authorId   Int?
  authorName String?  //how should I set this?
}
```

Thanks :lemon:

**Comment by ryands17** [comment, marked as answer]
@spidertwin2 👋 

I wouldn't suggest storing the `authorName` directly on the `Post` but fetching it from the `User` model as there is a directly relation from `Post` to `User`.

You can easily fetch the author name as follows:

```ts
await prisma.post.findMany({
  include: { author: true }
})
```

This will give you the author name for each post without duplicating the name on the `Post` model.

**Reply by spidertwin2** [reply]
Makes sense, I shifted `name` to `User` model.

Merci.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
